### ISSUE TITLE
NOT NULL in triggers/functions

### ISSUE THREAD

**Issue by BranislavBajuzik** [3YOURMIND/django-migration-linter#263]
Hello.

I am using this tool to lint my Postgres migrations (definitely prevented several headaches on deploy) and I found a false-positive:

```sql
BEGIN;
-- Create trigger api_user_update on model user
            CREATE OR REPLACE FUNCTION "public"._pgtrigger_should_ignore(
                trigger_name NAME
            )
            RETURNS BOOLEAN AS $$
                DECLARE
                    _pgtrigger_ignore TEXT[];
                BEGIN
                    IF _pgtrigger_ignore IS NOT NULL THEN
                        SELECT trigger_name = ANY(_pgtrigger_ignore)
                 -- And so on
COMMIT;
```

The migration/sql is courtesy of [django-pgtrigger](https://django-pgtrigger.readthedocs.io/en/4.7.0/).

This gets flagged because of the if statement, which I think should not trip up the detection. It gets flagged here: https://github.com/3YOURMIND/django-migration-linter/blob/3503367f1fcb34e633577cf70df9ab037d1f394c/django_migration_linter/sql_analyser/base.py#L43

Some of my hand-written migrations also get flagged for this reason.

 If I also may, I suggest that the `.startswith()` checks should be `.strip().lower().startswith()` to deal with possible future regressions because of indented/hand-written migrations

**Comment by David-Wobrock** [comment]
Hey @BranislavBajuzik 

Thanks for taking interesting in the library 🙏 

Indeed, the checks are a bit raw sometimes. Feel free to suggest a PR! 😄 
Else I can try to find some times in the next weeks/months

**Comment by kekekekule** [comment]
@BranislavBajuzik @David-Wobrock I've suggested a PR here: https://github.com/3YOURMIND/django-migration-linter/pull/273

**Comment by David-Wobrock** [comment]
Addressed in https://github.com/3YOURMIND/django-migration-linter/pull/282.
